generator/src/main/java/com/scottlogic/deg/generator/generation/ChecksummedCodeStringGenerator.java

```diff
@@ -78,31 +146,46 @@ public long getValueCount() {
                     generateRandomInvalidCheckDigitStrings(randomNumberGenerator)),
                 randomNumberGenerator);
         }
-        return new ProjectingIterable<>(sansCheckDigitGenerator.generateRandomValues(randomNumberGenerator),
-            sansCheckDigit -> sansCheckDigit + calculateCheckDigit(sansCheckDigit));
+        return new FilteringIterable<>(
```

Comment:
This part is repeated 3 times - it would be good to extract it out to a common method.